x-pack/plugins/reporting/server/routes/common/generate/request_handler.ts

```diff
@@ -136,14 +197,16 @@ export class RequestHandler {
     let report: Report | undefined;
     try {
       report = await this.enqueueJob(exportTypeId, jobParams);
+      const { basePath } = this.reporting.getServerInfo();
+      const publicDownloadPath = basePath + PUBLIC_ROUTES.JOBS.DOWNLOAD_PREFIX;
+
       // return task manager's task information and the download URL
-      const downloadBaseUrl = getDownloadBaseUrl(this.reporting);
       counters.usageCounter();
 
       return this.res.ok({
         headers: { 'content-type': 'application/json' },
         body: {
-          path: `${downloadBaseUrl}/${report._id}`,
+          path: `${publicDownloadPath}/${report._id}`,
```

Comment:
FIXME: this should use the `internal` path when the request context is internal.